### DISCUSSION TITLE
Does Nx segment require a workspace.json

### DISCUSSION THREAD

**Discussion by gabrielnemeth** [JanDeDobbeleer/oh-my-posh#2655] [Troubleshoot]
I have a Nx monorepo (Angular/Nest), however it does not have a `workspace.json`. I'm asking this because the segment is not displayed, and I'm trying to figure out why. I use the example config from docs:
```
{
  "type": "nx",
  "style": "powerline",
  "powerline_symbol": "\uE0B0",
  "foreground": "#000000",
  "background": "#1976d2",
  "template": " \uE753 {{ .Full }} "
}
```

My posh version is the latest 8.29.5, and I'm on Win 11 / Powershell 7.

**Comment by JanDeDobbeleer** [comment]
@gabrielnemeth correct, it needs a `workspace.json` as the default setting for `display_mode` is `files`. Setting that to `always` will display the segment at all times. Is there perhaps another file that could be used to identify a monorepo?

**Reply by gabrielnemeth** [reply]
I guess the `nx.json` file. The `workspace.json` is not always present. For e.g., if you generate an Angular app, it uses `angular.json` instead of `workspaces.json`.

**Reply by JanDeDobbeleer** [reply]
@gabrielnemeth we can extend the logic for sure. Allow me to adjust that after my holidays.

**Reply by gabrielnemeth** [reply]
Thank you ;) Have a great holiday!

**Reply by JanDeDobbeleer** [reply]
See https://github.com/JanDeDobbeleer/oh-my-posh/pull/2656
